src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,218 @@
+
+/**
+ * Class for TaskList components of Duke.
+ */
+public class TaskList {
+    /**
+     * Class for a singular Task.
+     */
+    class Task {
+        protected String description;
+        protected boolean isDone;
+
+        public Task(String description) {
+            this.description = description;
+            this.isDone = false;
+        }
+
+        public boolean getIsDone() {
+            return isDone;
+        }
+
+        public String getDescription() {
+            return description;
+        }
+
+        public void mark() {
+            this.isDone = true;
+        }
+
+        public void unmark() {
+            this.isDone = false;
+        }
+
+        public String getSavedString() {
+            return String.format("T | %s | %s\n",this.isDone, this.description);
+        }
+
+        @Override
+        public String toString() {
+            return String.format("[T] %s %s", Ui.checkbox(this.getIsDone()), this.getDescription());
+        }
+    }
+
+    /**
+     * Class for a singular Deadline.
+     */
+    class Deadline extends Task {
+        String deadline;
+        LocalDate date;
+
+        public Deadline(String description, String deadline) {
+            super(description);
+            this.deadline = deadline;
+        }
+
+        public Deadline(String description, LocalDate date) {
+            super(description);
+            this.date = date;
+        }
+
+        @Override
+        public String getSavedString() {
+            return String.format("D | %s | %s | %s\n",this.getIsDone(), this.getDescription(), this.deadline);
+        }
+
+        @Override
+        public String toString() {
+            return String.format("[D] %s %s (by: %s)", Ui.checkbox(this.getIsDone()), this.getDescription(), this.deadline);
+        }
+    }
+
+    /**
+     * Class for a singular Event.
+     */
+    class Event extends Task {
+        String time;
+
+        public Event(String description, String time) {
+            super(description);
+            this.time = time;
+        }
+
+        @Override
+        public String getSavedString() {
+            return String.format("E | %s | %s | %s\n",this.getIsDone(), this.getDescription(), this.time);
+        }
+
+        @Override
+        public String toString() {
+            return String.format("[E] %s %s (at: %s)", Ui.checkbox(this.getIsDone()), this.getDescription(), this.time);
+        }
+    }
+
+    int pointer = 0;
+    protected ArrayList<Task> list;
+    /**
+     * Constructor for TaskList.
+     */
+    public TaskList() {
+        this.list = new ArrayList<>(100);
+    }
+
+    /**
+     * Constructor for TaskList.
+     * @param taskList - tasklist to copy data from
+     */
+    public TaskList(TaskList taskList) {
+        this.list = taskList.list;
+    }
+
+    /**
+     * Adds a task to the list.
+     * @param description - description of the task
+     */
+    public Task addTask(String description) {

```

Comment:
Just another suggestion, but it might be better if there was a single generic addTask(Task t) method responsible for adding the various types of tasks (e.g. Event, Deadline) to make use of polymorphism, instead of having a separate method for adding each type of task